packages/lib/modules/pool/PoolDetail/PoolStats/PoolSnapshot/UserSnapshotValues.tsx

```diff
@@ -55,9 +62,30 @@ export function UserSnapshotValues() {
     }
   })
 
-  const tokens = currentRewardsPerWeek
-    .filter(reward => bn(reward.rewardPerSecond).gt(0))
-    .map(reward => getToken(reward.tokenAddress, chain)) as GqlToken[]
+  const tokens = useMemo(
```

Comment:
Same here, I would extract the logic to a new tested hook that receives `balRewards` and `nonBalRewards` and returns the `balanceMap`

Reply:
moved logic to new hook, no tests yet